### DISCUSSION TITLE
Rule for pointless assignments

### DISCUSSION THREAD

**Discussion by ChristophP** [eslint/eslint#17714] [Q&A] [answered]
I recently have to debug a lot of old code and had some need for an ESLint that tells me when an assignment is pointless. A code example:

```js
let name = "someval"; // pointless assignment, the value is never used

if (someComplicatedCondition) {
  name = "if";
} else {
  name = "else";
}
```
The following code would not violate the rule

```js
let name"; // no assignment

if (someComplicatedCondition) {
  name = "if";
} else {
  name = "else";
}
```

neither would this

```js
let name = "someval"; // pointless assignment
console.log(name); // here's a read before the variable is reassigned, so it's not pointless

if (someComplicatedCondition) {
  name = "if";
} else {
  name = "else";
}
```

In this example, the first assignment to `"someval"` is pointless, because the variable `name` is reassignment below without being read anywhere.

So I found myself wanting a rule that finds cases of values being assigned without being read before they are reassigned.
I looked around a bit but couldn't find anything like this. Does anyone know if a rule like this exists? If not, would that be considered a good addition?

PS: Keep in mind the example above is quite obvious but the codebase I'm dealing with cases like these are less obvious.

**Comment by mdjermanovic** [comment, marked as answer]
> So I found myself wanting a rule that finds cases of values being assigned without being read before they are reassigned.
> I looked around a bit but couldn't find anything like this. Does anyone know if a rule like this exists? If not, would that be considered a good addition?

It currently doesn't exist in the core, but a proposal for this rule has been accepted (https://github.com/eslint/eslint/issues/17559) and we are already working on it in https://github.com/eslint/eslint/pull/17625.

**Reply by ChristophP** [reply]
That sounds fantastic. Thank you for the info :-)
